app/src/main/java/org/example/uiController/UIController.java

```diff
@@ -1,71 +1,80 @@
+
 import java.util.Scanner;
 
 import org.example.common.Job;
 import org.example.queueManager.QueueManager;
+import org.example.dispatcher.Dispatcher;
 
 /**
  * The UI Controller is responsible for handling the user interface of the application.
  */
 public class UIController {
     private final Scanner userInput;
     private QueueManager job_queue = new QueueManager();
+    private Dispatcher dispatcher = new Dispatcher(job_queue);
+    private Thread dispatcherThread;
 
     public UIController(Scanner scanner) {
         this.userInput = scanner;
     }
-    public void generateUI(){
+
+    public void generateUI() {
         System.out.println("Welcome to the CSUBatch Scheduling Application");
         System.out.println("Thank you for downloading.");
         System.out.println("This System is meant to act as a scheduling application where jobs can be added to a queue that will be arranged based \n on the selected priority. \n commands: run, list, policy_change, help, exit");
         //userInteraction();
     }
-    public void userInteraction(){
+
+    public void userInteraction() {
         System.out.println("Please Enter Command: ");
         String command = userInput.nextLine();
         String[] commandarr = command.split(" "); //an array in place so the commands that are more than one word can be parsed.
-        while(!"exit".equals(command)){
-            if(commandarr[0].equals("run")){
-                if(commandarr.length != 4 ){
+        while (!"exit".equals(command)) {
+            if (commandarr[0].equals("run")) {
+                if (commandarr.length != 4) {
                     System.out.println("Invalid run command please try again");
                     break;
-                }
-                else{
-                    try { //Try in place to ensure that the user enters the run command format correctly andif not throws an error that does not crash the system.
+                } else {
+                    try { //Try in place to ensure that the user enters the run command format correctly and if not throws an error that does not crash the system.
                         String job_name = commandarr[1];
                         String job_time = commandarr[2];
                         String job_priority = commandarr[3];
                         int job_time_int = Integer.parseInt(job_time);
                         int job_priority_int = Integer.parseInt(job_priority);
+                        /* TODO: The Scheduler needs to be doing this work on the next two lines, not the UIController
+                           TODO: UIController just passes the params to the Scheduler */
                         Job userSubmitted = new Job(job_name, job_priority_int, job_time_int);
                         job_queue.enqueueJob(userSubmitted);
                         System.out.println("Job: " + job_name + " added to queue");
+
+                        // initialize dispatcher with a dedicated thread

```

Comment:
This is the main change, and the `startThread` and `endThread` methods. Everything else is file formatting.